presto-main/src/test/java/com/facebook/presto/operator/scalar/TestUnitFunctions.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.type.VarcharType;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+import static io.airlift.units.DataSize.Unit.BYTE;
+import static io.airlift.units.DataSize.Unit.GIGABYTE;
+import static io.airlift.units.DataSize.Unit.KILOBYTE;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static io.airlift.units.DataSize.Unit.PETABYTE;
+import static io.airlift.units.DataSize.Unit.TERABYTE;
+import static io.airlift.units.Duration.succinctDuration;
+import static io.airlift.units.Duration.succinctNanos;
+import static java.util.concurrent.TimeUnit.DAYS;
+import static java.util.concurrent.TimeUnit.HOURS;
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class TestUnitFunctions
+        extends AbstractTestFunctions
+{
+    private void testDuration(String sqlExpression, Duration expected)
+    {
+        assertFunction(sqlExpression, VarcharType.VARCHAR, expected.toString());
+    }
+
+    @Test
+    public void testSuccinctDuration()
+    {
+        testDuration("succinct_duration(123, 'ns')", succinctNanos(123));
```

Comment:
I understand that this test is testing the function implementation, not the implementation of `succinctDuration`. Nevertheless, I would prefer more straightforward test cases. Please change all tests here to use `assertFunction` directly.
```
testDuration("succinct_duration(123.45, 'ns')", succinctDuration(123.45, NANOSECONDS));
```
should instead say (Note that ... should be literals, without any logic)
```
assertFunction("succinct_duration(123.45, 'ns')", VARCHAR, utf8Slice("....."));
```

Reply:
Good point. This makes test cases more readable by showing the expected result strings. OK. I'll modify them.